compiler/util/files.cpp

```diff
@@ -638,6 +638,202 @@ void genIncludeCommandLineHeaders(FILE* outfile) {
   }
 }
 
+std::string genMakefileEnvCache(void) {
```

Comment:
Don't we already do this somewhere?

Reply:
It was a block of code in the original codegen_makefile() function, since I wanted to use it I made it a function itself.

Reply:
Ah, but I didn't see it removed from the original place and replaced with a call to this new function.  I may have just missed it, but it definitely should be done.